From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Manuel Traut <manuel.traut@mt.com>
Date: Tue, 17 Jan 2023 09:30:27 +0100
Subject: [PATCH] menu: Ignore prompt variable if timeout is != 0

Since 739e8361f3fe78038251216df6096a32bc2d5839, a system with the following
/boot/extlinux/extlinux.conf (which sets timeout to 50) immediately boots the
first entry in the config without displaying a boot menu.  According to the
description, that should only happen if both prompt and timeout are set to zero
in the config, but it also happens with timeout set to a non-zero value.

Reported-by: Karsten Merker <merker@debian.org>
Signed-off-by: Manuel Traut <manuel.traut@mt.com>
---
 common/menu.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/common/menu.c b/common/menu.c
index 94514177e4..d622bdebe3 100644
--- a/common/menu.c
+++ b/common/menu.c
@@ -279,6 +279,9 @@ int menu_get_choice(struct menu *m, void **choice)
 	if (!m->item_cnt)
 		return -ENOENT;
 
+	if (m->timeout)
+		return menu_interactive_choice(m, choice);
+
 	if (!m->prompt)
 		return menu_default_choice(m, choice);
 
-- 
2.40.0

